### ISSUE TITLE
nsis build fails randomly with unsupported optional native dependency

### ISSUE THREAD

**Issue by MasterOdin** [electron-userland/electron-builder#7029]
* **Electron-Builder Version**: 23.0.3

* **Node Version**: 16.16.0

* **Electron Version**: 13.6.9
* **Electron Type (current, beta, nightly)**: current

* **Target**: nsis

We recently added `msgpackr-extract` as a new native dependency of our application, though it's not supported. This has caused our nsis/windows builds to become flakey where it'll randomly fail, and just retriggering the pipeline is enough to get it to pass, so it's all the more confounding to try to debug.

A working build output will look like:
```
C:\Users\circleci\project\node_modules\.bin\electron-builder build --win --x64 --ia32 --publish onTag
  • electron-builder  version=23.0.3 os=10.0.17763
  • loaded configuration  file=package.json ("build" field)
  • rebuilding native dependencies  dependencies=keytar@7.8.0, sqlite3@4.2.0, msgpackr-extract@2.0.2, odbc@2.4.1, sqlite3@4.2.0 platform=win32 arch=ia32
  • install prebuilt binary  name=keytar version=7.8.0 platform=win32 arch=ia32 napi= 
  • rebuilding native dependency  name=sqlite3 version=4.2.0
  • rebuilding native dependency  name=odbc version=2.4.1
  • rebuilding native dependency  name=msgpackr-extract version=2.0.2
  • rebuilding native dependency  name=sqlite3 version=4.2.0
  • cannot build optional native dependency  name=msgpackr-extract
                                             version=2.0.2
                                             cause=fork/exec C:\Users\circleci\project\.yarn\releases\yarn-1.22.19.cjs: %1 is not a valid Win32 application.
                                             command='C:\Users\circleci\project\.yarn\releases\yarn-1.22.19.cjs' run install
                                             workingDir=C:\Users\circleci\project\packages\popsql-desktop\node_modules\@popsql\core\node_modules\msgpackr-extract
  • packaging       platform=win32 arch=ia32 electron=13.6.9 appOutDir=release\win-ia32-unpacked
  ...omitted lines
```

While a failing build will look like:
```
C:\Users\circleci\project\node_modules\.bin\electron-builder build --win --x64 --ia32 --publish onTag
  • electron-builder  version=23.0.3 os=10.0.17763
  • loaded configuration  file=package.json ("build" field)
  • rebuilding native dependencies  dependencies=keytar@7.8.0, sqlite3@5.0.10, msgpackr-extract@2.0.2, odbc@2.4.1, sqlite3@5.0.10 platform=win32 arch=x64
  • install prebuilt binary  name=keytar version=7.8.0 platform=win32 arch=x64 napi= 
  • rebuilding native dependency  name=sqlite3 version=5.0.10
  • rebuilding native dependency  name=odbc version=2.4.1
  • rebuilding native dependency  name=msgpackr-extract version=2.0.2
  • cannot build optional native dependency  name=msgpackr-extract
                                             version=2.0.2
                                             cause=fork/exec C:\Users\circleci\project\.yarn\releases\yarn-1.22.19.cjs: %1 is not a valid Win32 application.
                                             command='C:\Users\circleci\project\.yarn\releases\yarn-1.22.19.cjs' run install
                                             workingDir=C:\Users\circleci\project\packages\popsql-desktop\node_modules\@popsql\core\node_modules\msgpackr-extract
  • rebuilding native dependency  name=sqlite3 version=5.0.10
  ⨯ cannot execute  cause=fork/exec C:\Users\circleci\project\.yarn\releases\yarn-1.22.19.cjs: %1 is not a valid Win32 application.
                    command='C:\Users\circleci\project\.yarn\releases\yarn-1.22.19.cjs' run install
                    workingDir=C:\Users\circleci\project\packages\popsql-desktop\node_modules\@popsql\core\node_modules\odbc
```

where the `workingDir` of the "failing" workingDir will be randomly one of `keytar` or `sqlite3` or `odbc` without much pattern. My only thought is that there's some sort of race condition, but not even sure where I'd start looking in the codebase to figure out what's going on.

**Comment by MasterOdin** [comment]
Maybe this is related to the error affecting #6933?

**Comment by MasterOdin** [comment]
Closing this as it'll be fixed by the same thing as what'll fix #6933 (updating app-builder) from my tests.
